shared/constants/transaction.ts

```diff
@@ -196,4 +196,18 @@ export enum TokenStandard {
   none = 'NONE',
 }
 
-export const EIP712_PRIMARY_TYPE_PERMIT = 'Permit';
+/**
+ * EIP-712 Permit PrimaryTypes
+ */
+export enum PERMIT_PRIMARY_TYPE {
+  PERMIT = 'Permit',
+  PERMIT_BATCH = 'PermitBatch',
+  PERMIT_BATCH_TRANSFER_FROM = 'PermitBatchTransferFrom',
+  PERMIT_SINGLE = 'PermitSingle',
+  PERMIT_TRANSFER_FROM = 'PermitTransferFrom',
+  ORDER = 'Order',
+  ORDER_COMPONENTS = 'OrderComponents',
```

Comment:
Also on the values themselves, is our standard not `CamelCase` such as `OrderComponents = 'OrderComponents`?

Reply:
I changed it to a const, we could update if we change it back to an enum ([see comment](https://github.com/MetaMask/metamask-extension/pull/26107#discussion_r1691729879))